tt_metal/impl/trace/trace.cpp

```diff
@@ -15,14 +15,35 @@ namespace {
 static constexpr bool kBlocking = true;
 static constexpr bool kNonBlocking = false;
 
-size_t interleaved_page_size(
-    const uint32_t total_size, const uint32_t num_banks, const uint32_t min_size, const uint32_t max_size) {
-    // TODO #7110: Add logic to automatically figure out the trace buffer page size
-    // Calculate the page size for interleaved buffer that
-    // - is a power - of - 2 number (prefetch constraint),
-    // - AND is evenly distributed across the banks (start_page = 0)
-    // - AND falls inside min and max size constraints (optimal noc util),
-    return DeviceCommand::PROGRAM_PAGE_SIZE;
+// Min size is bounded by NOC transfer efficiency
+// Max size is bounded by Prefetcher CmdDatQ size
+static constexpr uint32_t kExecBufPageMin = 1024;
+static constexpr uint32_t kExecBufPageMax = 4096;
+
+// Assumes pages are interleaved across all banks starting at 0
+size_t interleaved_page_size(const uint32_t buf_size, const uint32_t num_banks, const uint32_t min_size, const uint32_t max_size) {
+    // Populate power of 2 numbers within min and max as candidates
+    vector<uint32_t> candidates;
+    for (uint32_t size = 1; size <= max_size; size <<= 1) {
+        if (size >= min_size) {
+            candidates.push_back(size);
```

Comment:
instantiate isn't exactly a perf path, but still would like to see fewer mallocs.  generate at init?  or size the vector once?  or leave as is...

Reply:
agreed i thought about using fancier bitwise operation tricks for this function but opted for the more readable version because it's not on the perf path

there is probably merit to making this a common function outside of trace class so it's useful in other places, maybe when we do that we can make it more efficient with things like pre-sizing and caching

Reply:
actually it's not that difficult just size the vector to roughly ~log2(max_size) should do the trick
```
    candidates.resize(32 - __builtin_clz(max_size));
```
I'll add it